addons/portal/static/src/chatter/core/attachment_upload_service_patch.js

```diff
@@ -0,0 +1,22 @@
+import { AttachmentUploadService } from "@mail/core/common/attachment_upload_service";
+
+import { patch } from "@web/core/utils/patch";
+
+patch(AttachmentUploadService.prototype, {
+    _buildFormData(formData, tmpURL, thread, composer, tmpId, options) {
+        super._buildFormData(...arguments);
+        const file = formData.get("ufile");
+        if (thread.model !== "discuss.channel") {
+            formData.append("name", file.name);
+            formData.append("file", file);
+        }
+        return formData;
```

Comment:
The portal chatter bundle is included in the frontend bundle and is completely separate from the backend bundle. So any change to the core or whatever in this bundle won't affect the backend chatter (a lot of tests would fail otherwise). It could only affect livechat, which I tried to prevent by filtering threads by `discuss.channel`. 
And yes, with `_get_thread_with_access` I think it's now possible to use `/mail/attachment/upload`.